pkg/proxy/util/utils.go

```diff
@@ -56,3 +59,45 @@ func ShouldSkipService(svcName types.NamespacedName, service *api.Service) bool
 	}
 	return false
 }
+
+type NetlinkHandle interface {
+	AddrAdd(link netlink.Link, addr *netlink.Addr) error
+	AddrDel(link netlink.Link, addr *netlink.Addr) error
+	LinkByName(dev string) (netlink.Link, error)
+}
+
+// EnsureAddressBind checks if address is bound to the interface and, if not, binds it. If the address is already bound, return true.
+func EnsureAddressBind(address, devName string, h NetlinkHandle) (exist bool, err error) {
+	dev, err := h.LinkByName(devName)
+	if err != nil {
+		return false, fmt.Errorf("error get interface: %s, err: %v", devName, err)
+	}
+	addr, err := netlink.ParseAddr(address + "/32")
+	if err != nil {
+		return false, fmt.Errorf("error parse ip address: %s/32, err: %v", address, err)
+	}
+	if err := h.AddrAdd(dev, addr); err != nil {
+		// "EEXIST" will be returned if the address is already bound to device
+		if err == syscall.Errno(syscall.EEXIST) {
+			return true, nil
+		}
+		return false, fmt.Errorf("error bind address: %s to interface: %s, err: %v", address, devName, err)
+	}
+	return false, nil
+}
+
+// UnbindAddress unbind address with the interface
+func UnbindAddress(address, devName string, h NetlinkHandle) error {
+	dev, err := h.LinkByName(devName)
+	if err != nil {
+		return fmt.Errorf("error get interface: %s, err: %v", devName, err)
+	}
+	addr, err := netlink.ParseAddr(address + "/32")
```

Comment:
Same as above, remove /32?

Reply:
No, If `/32` removed, an error of `invalid CIDR address`  will be thrown  by `netlink.ParseAddr`
I will add suitable mask or prefix by judging the address type.